Refactored code to activate evaluation

wahleric opened the pull request:
Removed auto-activation of evaluation upon creation of a FeatureSuggestion object. Now, listeners will be created and evaluation will start only when FeatureSuggestion.start() is called. Frontend clients can stop evaluation and remove any listeners created by calling FeatureSuggestion.stop()

Commit:
Refactored code to activate evaluation

DavidThien commented on backend_plugin/src/activation/Startup.java:
```diff
@@ -19,7 +18,8 @@ public void run() {
 					// Once we are loaded, make a new FeatureSuggestion
 					// Need to determine how FE will get the FeatureSuggestion object
-					new FeatureSuggestion();
+					FeatureSuggestion fs = new FeatureSuggestion();
```
If we aren't doing anything between construction and calling `start()`, then it's might worth it make the constructor start the service as well. I know this file will go through a lot of change as we're figuring out the interface with the front end, so it's probably fine for now, but something to think about.

DavidThien commented on backend_plugin/src/evaluators/Evaluator.java:
```diff
@@ -51,10 +53,12 @@ private void initializeListeners(IEditorPart editorWindow) {
 		// Get the document stored inside the editor window
 		ITextEditor editor = (ITextEditor) editorWindow;
```
I'm a little nervous of this cast without checking if `editor isinstanceof ITextEditor`. Also, since this method seems to work exclusively on `ITextEditors`, it's probably worth it to do this cast in the constructor an then make `initializeListeners` take an `ITextEditor` unless you anticipate this method being called from inside some native eclipse handler or somewhere else that can only pass an IEditorPart.

Commit:
Change `java.lang.String` to `String`

I had originally included this because I was having some weird dynamic loading error where another String type was being imported on eclipse launch, but this can be removed now because the problem is fixed moving to java 8.

DavidThien reviewed (changes requested):
Looks good, except for the casting comment.

johnbarUW commented on backend_plugin/src/evaluators/EvaluatorManager.java:
```diff
@@ -94,4 +72,41 @@ public void addEvaluator(IEditorPart editorWindow) {
 	}
+
+	public void start() {
+		for (IWorkbenchWindow window : PlatformUI.getWorkbench().getWorkbenchWindows()) {
```
Is it possible to add some inline comments on some of this? I can kind of guess what is going on, but having something short like "Grab all open documents" "Look at all windows in Eclipse", something that gives just a bit more context

johnbarUW commented on backend_plugin/src/evaluators/EvaluatorManager.java:
```diff
@@ -94,4 +72,41 @@ public void addEvaluator(IEditorPart editorWindow) {
+	 * Stops all Evaluators and removes any window or document listeners
+	 */
+	public void stop() {
+		for (IWorkbenchPage workbenchPage : this.openWindowListeners.keySet()) {
```
Same here in the stop method.  Something like "cycle through all workbenches and remove any listeners".

johnbarUW reviewed (changes requested):
I agree with David's comment about the cast without checking first.

Also, a few inline comments would be super helpful with the listener set up / tear down.

Commit:
Fixed casting issue during initialization

Commit:
Merge branch 'FS_start_stop' of github.com:DavidThien/IDE-IT into FS_start_stop

Commit:
Refactored Evaluator creation from EvaluatorManager

wahleric commented:
Instead of checking if the IEditorPart is an instance of ITextEditor inside the Evaluator constructor, I've refactored things a bit so that the Evaluator constructor itself takes an ITextEditor rather than an IEditorPart. This way, we never end up creating a useless Evaluator for a non-text-editor window.

Commit:
Removed unnecessary instance check

johnbarUW reviewed (approved):
Looks good. Thanks for adding those comments. It's much more readable now
